src/Trace/Trace.php

```diff
@@ -46,29 +45,27 @@ class Trace implements \Serializable
     private $traceId;
 
     /**
-     * @var TraceSpan[] List of TraceSpans to report
+     * @var Span[] List of Span to report
      */
     private $spans = [];
 
     /**
      * Instantiate a new Trace instance.
      *
-     * @param ConnectionInterface $connection The connection to Stackdriver Trace.
      * @param string $projectId The id of the project this trace belongs to.
      * @param string $traceId [optional] The id of the trace. If not provided, one will be generated
      *        automatically for you.
-     * @param array $spans [optional] Array of TraceSpan constructor arguments. See
-     *        {@see Google\Cloud\Trace\TraceSpan::__construct()} for configuration details.
+     * @param array $spans [optional] Array of Span constructor arguments. See
+     *        {@see Google\Cloud\Trace\Span::__construct()} for configuration details.
      * }
      */
-    public function __construct(ConnectionInterface $connection, $projectId, $traceId = null, $spans = null)
+    public function __construct($projectId, $traceId = null, $spans = null)
```

Comment:
If you change `$spans = null` to `array $spans = []`, you could eliminate the conditional on line 66 -- an empty array can be given to `array_map`, and the callable will never be invoked.